src/v/cluster/partition.h

```diff
@@ -86,6 +86,16 @@ class partition {
         return _raft->make_reader(std::move(config), deadline);
     }
 
+    /**
+     * The first index at which a consumer could read from. This will differ
```

Comment:
Can you give an example of when the start_offset from a retention-driven prefix truncation would differ from the visible_start_offset from a DeleteRecords?  It's not obvious to me why they need to be separate concepts, couldn't we just have one start offset that means "data below this is not readable", and use it for both cases?

Reply:
@mmaslankaprv would there be a valid need to be able to read from before the newly proposed `visible_start_offset()` ? If theres not, then Johns suggestion makes sense and I should just modify how `start_offset()` works

Reply:
This has been implemented: https://github.com/redpanda-data/redpanda/compare/187597f7ca3963a1003a4c70c36cfb942c5d535b..1189f7cc50cde38a366c9317a6b9ed3ea298c8d8